process/block/preprocess/scheduledTxsExecution_test.go

```diff
@@ -382,4 +391,625 @@ func TestScheduledTxsExecution_executeShouldWork(t *testing.T) {
+
+	scheduledTxsExec, _ := NewScheduledTxsExecution(
+		&testscommon.TxProcessorMock{},
+		&mock.TransactionCoordinatorMock{},
+		&genericMocks.StorerMock{},
+		&marshal.GogoProtoMarshalizer{},
+		&mock.ShardCoordinatorStub{},
+	)
+
+	rootHash := []byte("root hash")
+	gasAndFees := scheduled.GasAndFees{}
+	mapSCRs := map[block.Type][]data.TransactionHandler{
+		0: {
+			&transaction.Transaction{Nonce: 1},
+			&transaction.Transaction{Nonce: 2},
+		},
+		1: {
+			&transaction.Transaction{Nonce: 3},
+			&transaction.Transaction{Nonce: 4},
+		},
+	}
+
+	scheduledTxsExec.SetScheduledRootHashSCRsGasAndFees(
+		rootHash, mapSCRs, gasAndFees,
+	)
+
+	assert.Equal(t, rootHash, scheduledTxsExec.GetScheduledRootHash())
+	assert.Equal(t, gasAndFees, scheduledTxsExec.GetScheduledGasAndFees())
+	assert.Equal(t, mapSCRs, scheduledTxsExec.GetScheduledSCRs())
+}
+
+func TestScheduledTxsExecution_Setters(t *testing.T) {
+	t.Parallel()
+
+	rootHash := []byte("root hash")
+	gasAndFees := scheduled.GasAndFees{}
+
+	scheduledTxsExec, _ := NewScheduledTxsExecution(
+		&testscommon.TxProcessorMock{},
+		&mock.TransactionCoordinatorMock{},
+		&genericMocks.StorerMock{},
+		&marshal.GogoProtoMarshalizer{},
+		&mock.ShardCoordinatorStub{},
+	)
+	scheduledTxsExec.SetTransactionCoordinator(&mock.TransactionCoordinatorMock{})
+	scheduledTxsExec.SetTransactionProcessor(&testscommon.TxProcessorMock{})
+
+	scheduledTxsExec.SetScheduledGasAndFees(gasAndFees)
+	assert.Equal(t, gasAndFees, scheduledTxsExec.GetScheduledGasAndFees())
+
+	scheduledTxsExec.SetScheduledRootHash(rootHash)
+	assert.Equal(t, rootHash, scheduledTxsExec.GetScheduledRootHash())
+}
+
+func TestScheduledTxsExecution_getScheduledRootHashSCRsGasAndFeesForHeaderShouldFail(t *testing.T) {
+	t.Parallel()
+
+	rootHash := []byte("root hash")
+
+	t.Run("failed to get SCRs saved data from storage", func(t *testing.T) {
+		expectedErr := errors.New("storer err")
+		scheduledTxsExec, _ := NewScheduledTxsExecution(
+			&testscommon.TxProcessorMock{},
+			&mock.TransactionCoordinatorMock{},
+			&storageMocks.StorerStub{
+				GetCalled: func(_ []byte) ([]byte, error) {
+					return nil, expectedErr
+				},
+			},
+			&marshal.GogoProtoMarshalizer{},
+			&mock.ShardCoordinatorStub{},
+		)
+
+		_, _, _, err := scheduledTxsExec.getScheduledRootHashSCRsGasAndFeesForHeader(rootHash)
+		assert.Equal(t, expectedErr, err)
+	})
+	t.Run("failed to unmarshal data", func(t *testing.T) {
+		expectedErr := errors.New("marshaller err")
+		scheduledTxsExec, _ := NewScheduledTxsExecution(
+			&testscommon.TxProcessorMock{},
+			&mock.TransactionCoordinatorMock{},
+			&storageMocks.StorerStub{
+				GetCalled: func(_ []byte) ([]byte, error) {
+					return nil, nil
+				},
+			},
+			&testscommon.MarshalizerStub{
+				UnmarshalCalled: func(_ interface{}, _ []byte) error {
+					return expectedErr
+				},
+			},
+			&mock.ShardCoordinatorStub{},
+		)
+
+		_, _, _, err := scheduledTxsExec.getScheduledRootHashSCRsGasAndFeesForHeader(rootHash)
```

Comment:
could have checked that the other parameters returned are nil/default value

Reply:
done. checked also nil values